### PR TITLE
Minor suggestions for CIGAR errors & pos parsing.

### PR DESCRIPTION
**Pull request by dlaehnemann**
Before deleting the CigarError and CigarString parsing from my libprosic branch, I went over the implementation here and have some minor suggestions to @johanneskoester's implementation here. Mainly:

1. A distinction between `Unsupported` and `Unexpected` Cigar operations.
2. A mixture of our wordings in the CigarError messages.
3. The question whether to support the `Pad` operation, where the SAMv1 spec is not really clear on how it works and where I am not sure if any software uses it...

### CODE REVIEW

**Comment by dlaehnemann** [comment]
Just saw from the coveralls-check: proper tests are still missing, something I could model around my tests in libprosic. I'll add these to this pull request tomorrow, so feel free to ignore it until then.

**Comment by dlaehnemann** [comment]
Well, writing and including the tests came up with some corner cases on HardClips where the code did not yet enforce the SAM specification. Also, I found some recent (incoming into the spec master branch) clarifications of how the "Pad" Cigar operation works and implemented this accordingly. Basically, the existing wording that it reflects a "silent deletion from a padded reference" means that it consumes neither read nor reference sequence, which apparently can make sense in cleanly representing multiple alignments with insertions of varying size at the same position...

Also, I added a TODO whether we want to deprecate the `Back` operation in `CigarStringView::end_pos` as well--this would mean changing its return type to pass up the Error. And I changed the wording of the `include_del` parameter of `CigarStringView::read_pos` to reflect the current implementation, but am unsure if this is the intended behaviour.

Please Review away...

File: src/bam/record.rs [hunk 1]
```diff
@@ -606,7 +611,14 @@ impl CigarStringView {
...
                 &Cigar::Equal(l) | &Cigar::Diff(l) => pos += l as i32,
-                &Cigar::Back(l) => pos -= l as i32,
+                &Cigar::Back(l) => pos -= l as i32, // TODO: Don't we want to consistently deprecate this operation? E.g. by returning the following error (this would require a change of this function's signature!)
+                /*
+                {
+                    return Err(CigarError::UnsupportedOperation(
+                        "'back' (B) operation is deprecated according to htslib/bam_plcmd.c and is not in SAMv1 spec".to_owned()
+                    ));
+                }
+                */
```
Comment (johanneskoester) [inline src/bam/record.rs]: Yes, please do that.

**Comment by johanneskoester** [comment]
Very nice! If you would like to deprecate back in end_pos, that would be nice!

**Comment by dlaehnemann** [comment]
Ok, the deprecation of the `Back` Cigar operation is in `CigarStringView::end_pos()`. We have to watch out to make the respective changes in `libprosic` when using the next `rust-htslib` release.
